### ISSUE TITLE
Checklines are not deletable

### ISSUE THREAD

**Issue by Arthur-D** [supertuxkart/stk-editor#10]
Try creating a checkline, then select one of the points. You can move the points, but not delete them, and if you create a new track, everything except the checkline(s) will be removed. To get around that you'll have to restart the editor.

**Comment by ghost** [comment]
one of the main devs should try to reproduce this since the last pr, but this should be able to be closed now.

**Comment by Arthur-D** [comment]
The checklines are now not kept when creating a new track, but I still don't seem to be able to delete checklines. I try selecting both points but pressing Delete does not remove them.

**Comment by Arthur-D** [comment]
Huh now it seems to work, after reopening the track. Might be I goofed up when testing. :s

**Comment by vampy** [comment]
@Arthur-D does it work now for you?

**Comment by Arthur-D** [comment]
Not really, there is still some weirdness going on. First time opening it again, it did not work. Second time, it worked. Third time, the editor crashed, but I didn't run it in gdb so nothing useful there. Retrying several times, it worked all the time even if I had closed the editor, reopened it and then opening it for the first time in that session. So it usually works it seems like, but other times just stops working or even crashes.

Also, I am using @GaryShearer fork which has not been merged anyway, so even if it mostly works with it I would guess that the trunk version doesn't.

**Comment by GaryShearer** [comment]
@Arthur-D, can you provide the steps to reproduce this?

**Comment by Arthur-D** [comment]
1. I start the editor.
2. I select the test4 track I have sitting around.
3. I locate the checkline and zoom in on it.
4. I select the first endpoint and then shift-click on the other end.
5. I press delete, usually on the keyboard but at times I use the button.

Just tried it again but could not reproduce it this time. Very frustrating that it's not appearing every time but still often enough to be an issue. :(
[test4.zip](https://github.com/supertuxkart/stk-editor/files/453403/test4.zip)

**Comment by psypherium** [comment]
I am also experiencing this issue deleting checklines, moving focus to another window and then returning to the editor will sometimes fix it.

Also, DVORAK keyboard layouts are not detected, but moving focus to another window and then returning to the editor will sometimes fix it.

SDL?

**Comment by STK-Diogenes** [comment]
Hello Arthur-D and hello psypherium, :-)

maybe I can help you a little bit.
Now, I have download your test4-track and tryd to delete the checkline, but, it have not can done.
What I can done is, the checkline bringing in the right position and work now.
Many Problems make the textures for road and how much objects pro map posible.
But also many objects are not insertable.
Now, I have a litte bit edited your track, for giving a exemble, but not with a texture for the road.
I hope now, that you like a bit this exemble and also, that I have can a little bit help you.

Greeting from Germany... ;-)
[test4.zip](https://github.com/supertuxkart/stk-editor/files/2373135/test4.zip)
